microprofile/fault-tolerance-smallrye/executor/src/main/java/org/wildfly/extension/microprofile/faulttolerance/MiniContextPropagation.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright 2020 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.extension.microprofile.faulttolerance;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Miniature context propagation API, adapted from MicroProfile Context Propagation and SmallRye Context Propagation.
+ * @author Ladislav Thon (c) 2020 Red Hat, Inc.
+ * @author Emmanuel Hugonnet (c) 2020 Red Hat, Inc.
+ */
+public final class MiniContextPropagation {
```

Comment:
If this and other new classes can be package private that is better.

It's a private module but locking things down is better.